internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -1075,10 +1081,16 @@ func FlattenSiteConfigWindowsAppSlot(appSiteSlotConfig *web.SiteConfig, currentS
 
 	winAppStack := ApplicationStackWindows{}
 
-	winAppStack.NetFrameworkVersion = pointer.From(appSiteSlotConfig.NetFrameworkVersion)
-	if currentStack == CurrentStackDotNetCore {
-		winAppStack.NetCoreVersion = pointer.From(appSiteSlotConfig.NetFrameworkVersion)
+	if appSiteSlotConfig.NetFrameworkVersion != nil {
+		winAppStack.NetFrameworkVersion = pointer.From(appSiteSlotConfig.NetFrameworkVersion)
+		if *appSiteSlotConfig.NetFrameworkVersion == DotNetCLRVersionFourPointOh {
+			if !features.FourPointOhBeta() && metadata.ResourceData.Get("site_config.0.application_stack.0.dotnet_core_version").(string) != "" {
```

Comment:
I think we can do this without the metadata?In the pre-4.0 path we'll still have `currentStack == CurrentStackDotNetCore` to check this? 

e.g. 
```suggestion
			if !features.FourPointOhBeta() && currentStack == CurrentStackDotNetCore {
```

Reply:
updated